test/powershell/Modules/Microsoft.PowerShell.Utility/Export-Csv.Tests.ps1

```diff
@@ -43,81 +45,154 @@ Describe "Export-Csv" -Tags "CI" {
+        $results = Import-Csv -Path $testCsv
+
+        $results[0].P1 | Should -BeExactly "first"
+        $property = $results.PSObject.Properties.Name
+        $property | Should -BeExactly "P1"
     }
 
     It "Test basic function works well" {
-        $input = [pscustomobject]@{ "P1" = "V11"; "P2" = "V12"; "P3" = "V13" }
-        $input | Export-Csv -Path $filePath -NoTypeInformation
-        $results = Import-Csv $filePath
+        $in = [pscustomobject]@{ "P1" = "V11"; "P2" = "V12"; "P3" = "V13" }
+        $in | Export-Csv -Path $testCsv -NoTypeInformation
+        $results = Import-Csv -Path $testCsv
+
         $results.P1 | Should -BeExactly "V11"
         $results.P2 | Should -BeExactly "V12"
         $results.P3 | Should -BeExactly "V13"
     }
 
     It "Test if it works with special character" {
         $v3 = "abc" + $newLine + "foo"
-        $input = [pscustomobject]@{ "P1" = "  "; "P2" = "abc,foo"; "P3" = $v3}
-        $input | Export-Csv -Path $filePath -NoTypeInformation
-        $results = Import-Csv $filePath
+        $in = [pscustomobject]@{ "P1" = "  "; "P2" = "abc,foo"; "P3" = $v3}
+        $in | Export-Csv -Path $testCsv -NoTypeInformation
+        $results = Import-Csv -Path $testCsv
+
         $results.P1 | Should -BeExactly "  "
         $results.P2 | Should -BeExactly "abc,foo"
-        $results.P3 | Should -Be $v3
-    }
-
-    It "Test force switch works well" {
-        $input = [pscustomobject]@{ "P1" = "first" }
-        $input | Export-Csv -Path $filePath
-
-        $input =  [pscustomobject]@{ "P2" = "second" }
-        $input | Export-Csv -Path $filePath -Force
-        $results = Import-Csv $filePath
-
-        $results.P2 | Should -BeExactly "second"
-        $property = $results | Get-Member | Where-Object { $_.MemberType -eq "NoteProperty" } | ForEach-Object { $_.Name }
-        $property | Should -Not -Be P1
+        $results.P3 | Should -BeExactly $v3
     }
 
     It "Test export-csv with a useculture flag" {
-        $outputFilesDir = Join-Path $TestDrive -ChildPath "Monad"
-        $fileToGenerate = Join-Path $outputFilesDir -ChildPath "CSVTests.csv"
+        $outputFilesDir = Join-Path -Path $TestDrive -ChildPath "Monad"
+        $fileToGenerate = Join-Path -Path $outputFilesDir -ChildPath "CSVTests.csv"
         $delimiter = (Get-Culture).TextInfo.ListSeparator
         New-Item -Path $outputFilesDir -ItemType Directory -Force
-        Get-Item -Path $outputFilesDir| Export-Csv -Path $fileToGenerate -UseCulture -NoTypeInformation
+        Get-Item -Path $outputFilesDir | Export-Csv -Path $fileToGenerate -UseCulture -NoTypeInformation
         $contents = Get-Content -Path $fileToGenerate
+
         $contents.Count | Should -Be 2
         $contents[0].Contains($delimiter) | Should -BeTrue
         $contents[1].Contains($delimiter) | Should -BeTrue
```

Comment:
We could use `Should -Contain`

Reply:
No. If the delimiter is comma, Should treats input as array, so `$contents[0] | Should -Contain $delimiter` will fail.

Reply:
`$contents[0]` is a string not a language operator.

Reply:
```
It "Test export-csv with a useculture flag" {
    $outputFilesDir = Join-Path -Path $TestDrive -ChildPath "Monad"
    $fileToGenerate = Join-Path -Path $outputFilesDir -ChildPath "CSVTests.csv"
    $delimiter = (Get-Culture).TextInfo.ListSeparator
    New-Item -Path $outputFilesDir -ItemType Directory -Force
    Get-Item -Path $outputFilesDir | Export-Csv -Path $fileToGenerate -UseCulture -NoTypeInformation
    $contents = Get-Content -Path $fileToGenerate

    $contents.Count | Should -Be 2
    $contents[0] | Should -Contain $delimiter
}


  [-] Test export-csv with a useculture flag 144ms
    Expected ',' to be found in collection "PSPath","PSParentPath","PSChildName","PSDrive","PSProvider","PSIsContainer","Mode","BaseName","Target","LinkType","Name","Parent","Exists","Root","FullName","Extension","CreationTime","CreationTimeUtc","LastAccessTime","LastAccessTimeUtc","LastWriteTime","LastWriteTimeUtc","Attributes", but it was not found.
    186:         $contents[0] | Should -Contain $delimiter
```

Reply:
Thanks for the repo! This is my misconception. Closed.